google/cloud/security/scanner/audit/instance_network_interface_rules_engine.py

```diff
@@ -227,28 +227,23 @@ def find_violations(self, instance_network_interface_list):
                 instance_network_interface.network)
             project = network_and_project.group(1)
             network = network_and_project.group(2)
-            is_external_network = (instance_network_interface.access_configs
-                                   is not None)
-            if not self.rules['whitelist'].get(project):
+            is_external_network = (instance_network_interface.access_configs is
+                                   not None)
+            ips = None
+            if is_external_network:
+                ips = [config.get('natIP', '') for config in
```

Comment:
nit: Not sure if this makes a difference, but do you think that this evaluation of ```ips``` would be better placed into the the next if statement?  So, that it's only evaluated when it directly needs to be yielded?

```
if (network not in self.rules['whitelist'].get(project, [])
    and is_external_network):
    ips = [config.get('natIP', '') for config in instance_network_interface.access_configs]
    yield self.RuleViolation(
        blah=blah,
        ips=ips)
```

Reply:
Also you can ensure only IPs are added to the list by adding an if to the comprehension:

ips = [config['natIP']
    for config in instance_network_interface.access_configs
    if 'natIP' in config]